crates/api_common/src/utils.rs

```diff
@@ -915,30 +915,48 @@ async fn proxy_image_link_internal(
 
 /// Rewrite a link to go through `/api/v4/image_proxy` endpoint. This is only for remote urls and
 /// if image_proxy setting is enabled.
-pub async fn proxy_image_link(link: Url, context: &LemmyContext) -> LemmyResult<DbUrl> {
-  proxy_image_link_internal(link, context.settings().pictrs()?.image_mode, context).await
+pub async fn proxy_image_link(
+  link: Url,
+  is_thumbnail: bool,
+  context: &LemmyContext,
+) -> LemmyResult<DbUrl> {
+  proxy_image_link_internal(
+    link,
+    context.settings().pictrs()?.image_mode,
+    is_thumbnail,
+    context,
+  )
+  .await
 }
 
 pub async fn proxy_image_link_opt_apub(
   link: Option<Url>,
   context: &LemmyContext,
 ) -> LemmyResult<Option<DbUrl>> {
   if let Some(l) = link {
-    proxy_image_link(l, context).await.map(Some)
+    proxy_image_link(l, false, context).await.map(Some)
   } else {
     Ok(None)
   }
 }
 
 fn build_proxied_image_url(
   link: &Url,
-  protocol_and_hostname: &str,
-) -> Result<Url, url::ParseError> {
-  Url::parse(&format!(
+  is_thumbnail: bool,
+  context: &LemmyContext,
+) -> LemmyResult<Url> {
+  let mut url = format!(
     "{}/api/v4/image/proxy?url={}",
-    protocol_and_hostname,
-    encode(link.as_str())
-  ))
+    context.settings().get_protocol_and_hostname(),
+    encode(link.as_str()),
+  );
+  if is_thumbnail {
+    url = format!(
+      "{url}&max_size={}",
```

Comment:
is this supposed to limit the source image size to avoid processing large images or the resulting thumbnail size?
if it's supposed to resize, shouldn't it be `resize=` instead? i don't see `max_size` in the pict-rs docs at all.